pkg/controller/azvolume.go

```diff
@@ -215,9 +215,10 @@ func (r *ReconcileAzVolume) triggerDelete(ctx context.Context, azVolume *azdiskv
 	volumeDeleteRequested := volumeDeleteRequested(azVolume)
 	preProvisionCleanupRequested := isPreProvisionCleanupRequested(azVolume)
 
-	mode := deleteCRIOnly
+	// primary attachments should be detached only if volume is being deleted or pv was deleted.
+	mode := cleanUpAttachmentForUninstall
 	if volumeDeleteRequested || preProvisionCleanupRequested {
-		mode = detachAndDeleteCRI
+		mode = cleanUpAttachment
```

Comment:
This if clause is now redundant to the one on line 232. This assignment can now be moved within the latter if clause.

Reply:
True